### PR TITLE
Display path to preloaded res + button to open it

### LINKED ISSUE #15556: ResourcePreloader panel lacks information on resources
**Godot version:**
Master (commit 2ac27608a6892d8d7e28d8afbbcc98505eb3e4b3)

**OS/device including version:**
Mageia 6

**Issue description:**
ResourcePreloader panel misses information on what resources are set in it. We only have a list of IDs and type of resource. If the resource is a scene, the type is `PackedScene` whatever the scene actually loaded. The only way to get more information is to hover each line with mouse to get the actual path to the resource. 
image

I think this panel has enough emtpy width space to add this information directly as a column, instead of having to hover.
A button to access the resource would be interesting too.

**Steps to reproduce:**
Create a scene
Add a ResourcePreloader node and select it
Add a resource in the bottom panel, choose another scene.

**Minimal reproduction project:**
N/A

### PR DESCRIPTION
**Pull request by StraToN**
Fixes #15556 
Before:
image

After:
image

### CODE REVIEW

**Comment by akien-mga** [comment]
Build fails with GCC, Clang and MSVC.. :) https://travis-ci.org/godotengine/godot/jobs/328780893

File: editor/plugins/resource_preloader_editor_plugin.cpp [hunk 1]
```diff
@@ -229,17 +229,32 @@ void ResourcePreloaderEditor::_update_library() {
...
-			ti->set_icon(1, get_icon(type, "EditorIcons"));
+			ti->set_icon(2, get_icon(type, "EditorIcons"));
 	}
 
 	//player->add_resource("default",resource);
 }
 
+void ResourcePreloaderEditor::_cell_button_pressed(Object *p_item, int p_column, int p_id) {
+
+	TreeItem *item = Object::cast_to<TreeItem>(p_item);
```
Comment (vnen) [inline editor/plugins/resource_preloader_editor_plugin.cpp]: If this is always a `TreeItem`, shouldn't the method signature be typed as `Ref<TreeItem>` instead?

In any case, I would an `ERR_FAIL_COND` to be on the safe side.

Comment (StraToN) [reply editor/plugins/resource_preloader_editor_plugin.cpp]: > If this is always a `TreeItem`, shouldn't the method signature be typed as `Ref<TreeItem>` instead?
I am not certain to understand. I took the example from scene_tree_editor.cpp (https://github.com/godotengine/godot/blob/34c988cfa92f19c232b65990704816ba1c7d2622/editor/scene_tree_editor.cpp#L48) and made it similar.

Ok, I put the `ERR_FAIL_COND` back.

Comment (StraToN) [reply editor/plugins/resource_preloader_editor_plugin.cpp]: Back to your remark.
I don't know how I can cast the input of `_cell_button_pressed()` *before* it is called. It is connected to a `button_pressed` signal (here: https://github.com/godotengine/godot/pull/15713/files#diff-1704455f99063b3e8f3e99d9b2364720R406) and I looked at many places in the whole project and every callback function takes the same parameters.

Comment (vnen) [reply editor/plugins/resource_preloader_editor_plugin.cpp]: Oh, so the `button_pressed` signature has an `Object*`. It's weird to me, but such change would be out of the scope of this PR, so it's okay the way it is.

**Pull request merged**